internal/civisibility/utils/git.go

```diff
@@ -30,8 +39,75 @@ type localGitData struct {
 	CommitMessage  string
 }
 
-// regexpSensitiveInfo is a regular expression used to match and filter out sensitive information from URLs.
-var regexpSensitiveInfo = regexp.MustCompile("(https?://|ssh?://)[^/]*@")
+var (
+	// regexpSensitiveInfo is a regular expression used to match and filter out sensitive information from URLs.
+	regexpSensitiveInfo = regexp.MustCompile("(https?://|ssh?://)[^/]*@")
+
+	// isGitFoundValue is a boolean flag indicating whether the Git executable is available on the system.
+	isGitFoundValue bool
+
+	// gitFinder is a sync.Once instance used to ensure that the Git executable is only checked once.
+	gitFinder sync.Once
+)
+
+// isGitFound checks if the Git executable is available on the system.
+func isGitFound() bool {
+	gitFinder.Do(func() {
+		_, err := exec.LookPath("git")
+		isGitFoundValue = err == nil
+		if err != nil {
+			log.Debug("civisibility.git: git executable not found")
+		}
+	})
+	return isGitFoundValue
+}
+
+// execGit executes a Git command with the given arguments.
+func execGit(args ...string) ([]byte, error) {
+	if !isGitFound() {
+		return nil, errors.New("git executable not found")
+	}
+	return exec.Command("git", args...).CombinedOutput()
+}
+
+// execGitString executes a Git command with the given arguments and returns the output as a string.
+func execGitString(args ...string) (string, error) {
+	out, err := execGit(args...)
+	strOut := strings.TrimSpace(strings.Trim(string(out), "\n"))
+	if err != nil {
+		return strOut, err
+	}
```

Comment:
This is the same whether the error happens or not :smile: (the same applies to the function below)

Reply:
🤦🏻 I must be tired. thanks!